### PR TITLE
Modernizing

### PR DESCRIPTION
**Pull request by TehShrike**
LevelUP upgrade + Promises + async/await.

I didn't bother modernizing most of the tests.

This is a precursor to upgrading [levelup-cache](https://github.com/TehShrike/levelup-cache/) to be tested with LevelUP 2, and also make it easier to use async/await to eliminate some race conditions in its use of expire-unused-keys.

### CODE REVIEW

**Comment by TehShrike** [comment]
@saibotsivad or @ArtskydJ want to review? <3

**Comment by TehShrike** [comment]
You can see the new breaking change list in the readme: https://github.com/TehShrike/expire-unused-keys/pull/10/files#diff-0730bb7c2e8f9ea2438b52e419dd86c9R13

File: readme.md [hunk 1]
```diff
@@ -8,81 +10,94 @@ To lessen the pain in this task, I need a library that will keep track of what t
...
+
+# Example
 
 ```js
-    // Some levelUP db
-	var db = level('wat')
-	// Expire stuff after 15 seconds of inactivity
-	var expirer = new Expirer(15000, db)
+// Expire stuff after 15 seconds of inactivity
+const expirer = new Expirer(15 * 1000, someLevelUpDb)
```
Comment (ArtskydJ) [inline readme.md]:
```js
const expirer = new Expirer({ timeoutMs: 15 * 1000, db: someLevelUpDb })
```

Comment (TehShrike) [reply readme.md]: oh snap!  That's what I get for not using jsmd.  I didn't want it to be waiting around for the timeouts, though...

Comment (ArtskydJ) [reply readme.md]: That's all I see from a very preliminary scan... Might be able to take a better look later.

**Pull request merged**